src/development/LocalTest/Controllers/Storage/InstancesController.cs

```diff
@@ -619,6 +619,47 @@ public async Task<ActionResult<Instance>> Delete(int instanceOwnerPartyId, Guid
             return updatedInstance;
         }
 
+        /// <summary>
+        /// Updates the data values on an instance.
+        /// </summary>
+        /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+        /// <param name="instanceGuid">The id of the instance to confirm as complete.</param>
+        /// <param name="dataValues">Collection of changes to the presentation texts collection.</param>
+        /// <returns>The instance that was updated with an updated collection of data values.</returns>
+        [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_WRITE)]
+        [HttpPut("{instanceOwnerPartyId:int}/{instanceGuid:guid}/datavalues")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [Consumes("application/json")]
+        [Produces("application/json")]
+        public async Task<Instance> UpdateDataValues(
+            [FromRoute] int instanceOwnerPartyId,
+            [FromRoute] Guid instanceGuid,
+            [FromBody] DataValues dataValues)
+        {
+            var instanceId = $"{instanceOwnerPartyId}/{instanceGuid}";
+            Instance instance = await _instanceRepository.GetOne(instanceId, instanceOwnerPartyId);
+
+            if (instance.DataValues == null)
+            {
+                instance.DataValues = new Dictionary<string, string>();
```

Comment:
instance.PresentationTexts ??= new Dictionary<string, string>();
Consider the shorthand here to save some code lines, and feel free to change this in the endpoint for presentationTextd as well.